### DISCUSSION TITLE
Unzipping data files during install

### DISCUSSION THREAD

**Discussion by deanjones** [python-poetry/poetry#6860] [Q&A]
Is it possible to specify that zipped data files in a project build are unzipped on install? I know I can do this using `setup.py` but not sure how to do this when using poetry to build, as it generates the `setup.py`. Is there a way to hook into the generated `setup.py` somehow? Or specify a script to run during installation? For various reasons, doing this at runtime is not an option (or at least, highly undesirable).

**Comment by clintonroy** [comment]
There's nothing like that in standard wheels, so poetry doesn't have an option for it. If you've already got your data files in zip format, I would leave them like that, and see if you can use the [importlib.resources](https://docs.python.org/3/library/importlib.html#module-importlib.resources) api to get files from them, without unzipping them yourself.

**Reply by deanjones** [reply]
Thanks for responding. The files are just gzipped, so it's not really a question of how to unzip them. I appreciate this is not really a poetry-issue, as it's not supported by wheels, but a post-install script to unzip the files is the best solution in this situation (it's technically possible to unzip the files at runtime, however security requirements mean that the runtime user does not have write permission to the data directory, which makes this not possible).

**Comment by neersighted** [comment]
This is not a Poetry-specific issue, but just a general request in the Python packaging ecosystem. There won't likely be a solution until it's solved ecosystem-wide, in the mean time, use of `importlib.resources` is probably your best best (you can access the files more or less transparently).

https://github.com/pypa/packaging-problems/issues/64
